src/qt/configurenamedialog.cpp

```diff
@@ -61,7 +64,17 @@ void ConfigureNameDialog::accept()
     returnData = ui->dataEdit->text();
     returnTransferTo = ui->transferTo->text();
 
-    QDialog::accept();
+    if(!IsValidJSON(data)){
+        QMessageBox::StandardButton MessageBoxInvalidJSON = 
+        QMessageBox::question(this, "Invalid JSON",
+                "Are you sure you want to continue anyway? The inputted JSON data is invalid.");
```

Comment:
Maybe tweak this message a bit to more explicitly indicate that continuing is likely to make the name unresolvable?

Reply:
Sure